tests/phpunit/tests/actions.php

```diff
@@ -309,6 +350,9 @@ function test_action_keyed_array() {
 
 	}
 
+	/**
+	 * @covers ::do_action
```

Comment:
```suggestion
	 * @covers ::remove_action
```

This test intends to test a one-shot circuit, i.e. a circuit where a callback can only be invoked one time. It's testing the registration process. How so? `do_action` loops through the registered callbacks. It will only invoke those callbacks that are registered to the event (tag) being fired.

In this case, the function being tested is the `remove_action` _when_ the callback gets invoked the first time. The callback could be directly invoked to test the functionality. But here it's using `do_action` as the invoking mechanism.

Note: 
This test needs work as It's not actually doing the test it intends to do. Why? `do_action` is called only once meaning the callback will only be called once. It needs to fire again to ensure the callback does not get invoked a second time. The test will need to be fixed in another ticket and patch. Adding to my list.